pkgs/by-name/bu/buildstream/package.nix

```diff
@@ -0,0 +1,69 @@
+{
+  python3Packages,
+  fetchFromGitHub,
+  lzip,
+  patch,
+  bubblewrap,
+  fuse3,
+  lib,
+  buildbox,
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "buildstream";
+  version = "2.4.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "apache";
+    repo = "buildstream";
+    tag = version;
+    hash = "sha256-6a0VzYO5yj7EHvAb0xa4xZ0dgBKjFcwKv2F4o93oahY=";
+  };
+
+  build-system = with python3Packages; [
+    setuptools
+    pdm-pep517
+    setuptools-scm
+    cython
+  ];
+
+  dependencies = with python3Packages; [
+    click
+    dulwich
+    grpcio
+    jinja2
+    markupsafe
+    packaging
+    pluginbase
+    protobuf
+    psutil
+    pyroaring
+    requests
+    ruamel-yaml
+    ruamel-yaml-clib
+    tomlkit
+    ujson
+  ];
+
+  buildInputs = [
+    buildbox
+    fuse3
+    lzip
+    patch
+  ];
+
+  nativeBuildInputs = with python3Packages; [
+    bubblewrap
+  ];
+
```

Comment:
```suggestion

  pythonImportsCheck = [ "buildstream" ];
  
  nativeCheckInputs = [
    pytestCheckHook
    versionCheckHook
  ];
  versionCheckProgramArg = "--version";

```

Reply:
> `versionCheckHook` works in my local checkout, but when `pytest` runs, I get a error with timeout args. I'll have to look at this tomorrow I think - I also had to put `python3Packages.pytestCheckHook`.

Yes, sorry I meant `python3Packages.pytestCheckHook`.
Maybe it also needs `python3Packages.pytest-timeout`.